rpc/src/main/kotlin/dk/cachet/carp/rpc/Main.kt

```diff
@@ -1,20 +1,33 @@
 package dk.cachet.carp.rpc
 
-import dk.cachet.carp.rpc.ApplicationServiceInfo.Companion.JSON_EXAMPLES_FOLDER
+import dk.cachet.carp.common.application.ApplicationServiceInfo
+import dk.cachet.carp.common.application.services.ApplicationService
 import org.apache.commons.io.FileUtils
+import org.reflections.Reflections
 import java.io.File
 
 
+/**
+ * Application service info for all CARP application services which are loaded on the current classpath.
+ */
+val applicationServices = Reflections( "dk.cachet.carp" )
+    .getSubTypesOf( ApplicationService::class.java )
+    .filter { it.isInterface }
+    .map { ApplicationServiceInfo( it ) }
+
+
 fun main( args: Array<String> )
 {
     // Create example requests for all request objects and responses of application service methods.
-    val exampleRequests = ApplicationServiceInfo.findInNamespace( "dk.cachet.carp" )
-        .associateWith { generateExampleRequests( it.serviceKlass, it.requestObjectKlass ) }
+    val exampleRequests = applicationServices
+        .associateWith { generateExampleRequests( it ) }
 
     // Output example requests.
-    FileUtils.deleteDirectory( JSON_EXAMPLES_FOLDER )
+    val jsonExamplesFolder = File( "build/rpc-examples/" )
+    FileUtils.deleteDirectory( jsonExamplesFolder )
```

Comment:
You probably want to catch `IOException` here

Reply:
I suspect this `rpc` application may become an entry point called as part of the Gradle build process to automate packaging test sources as part of releases.

If this fails, the build should fail. No clean way to handle the exception here for now. I think an exception by default results in an error code returned to the command line, which would be desired behavior. If the exception is unclear I can catch it to perhaps make it more meaningful, but will leave it as is for now.